src/main/java/com/course/fleet/management/api/controller/UserController.java

```diff
@@ -0,0 +1,72 @@
+package com.course.fleet.management.api.controller;
+
+import com.course.fleet.management.api.entity.User;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import net.datafaker.Faker;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+
+  private static final List<User> users;
+
+  static {
+    Faker faker = new Faker();
+    users =
+        IntStream.range(0, 10)
+            .mapToObj(
+                i -> {
+                  String firstName = faker.name().firstName();
+                  String lastName = faker.name().lastName();
+                  return new User(
+                      (long) i,
+                      firstName + " " + lastName,
+                      firstName + "." + lastName + "@mail.com");
+                })
+            .collect(Collectors.toList());
+  }
+
+  @GetMapping
+  public ResponseEntity<List<User>> getUsers() {
+    return ResponseEntity.ok(users);
+  }
+
+  @GetMapping("/{id}")
+  public ResponseEntity<Optional<User>> getUser(@PathVariable long id) {
+    Optional<User> userById = users.stream().filter(user -> user.getId() == id).findFirst();
```

Comment:
Una recomendación más avanzada para que tu código se vea más pro es:

Las variables que ya no vayas a reasignar con un = definelas como final, esto ya que ayudas a la optimización de tu código. Al no ser final la jvm lo observa constantemente por si se reasigna actualizar su ocupación en espacio de memoria. Al ser final le da el espacio de memoria que requiere y por así decirlo lo deja de observar, con esto es una responsabilidad menos que maneja.

```suggestion
    final Optional<User> userById = users.stream().filter(user -> user.getId() == id).findFirst();
```